concepts/error-handling/about.md

```diff
@@ -15,85 +14,79 @@ There are multiple ways to hendle errors in `Cairo`
 ## The Result Enum
 This approach handles errors gracefully, providing feedback insted of crashing or producing undefined behaviiour.
 
-**The code snippet below shows how enums could be defined:**
-```
-enum Result<T, E> {
-    Ok: T,
-    Err: E,
-}
-```
-
-> Let's Take an example of `Devide By Zero` 
-```
-fn divide(a: felt, b: felt) -> Result<felt, &'static str> {
++ Let's look at an example of `Divide By Zero` :
+```rust
+fn divide(a: felt252, b: felt252) -> Result<felt252, ByteArray> {
     if b == 0 {
         return Err("Error: Division by zero"); // Return an error if division by zero is attempted
     }
     Ok(a / b)
 }
-```
 
-```
-// Test with division by zero
-match divide(10, 0) {
-    Ok(result) => println!("Division successful: {}", result),
-    Err(e) => println!("{}", e),
+fn main() {
+    // Test with division by zero
+    match divide(10, 0) {
+        Ok(result) => println!("Division successful: {}", result),
+        Err(e) => println!("{}", e),
+    }
 }
 ```
 ## The `?` Operator
-`?` operator is used for minimilistic(concise) errors. Allow to write clean and readable code without explicit values.
+`?` operator is used for concise and implicit error handling, letting the calling function deal with any errors that might occur.
 
->Example
-```
-fn safe_divide(a: felt, b: felt) -> Result<felt, &'static str> {
++ Example
+```rust
+fn safe_divide(a: felt252, b: felt252) -> Result<felt252, ByteArray> {
     // The ? operator automatically generate the error if it occurs
     let result = divide(a, b)?;
     Ok(result)
 }
-```
 
-```
-match safe_divide(10, 0) {
-        Ok(value) => println!("Division successful: {}", value),
-        Err(e) => println!("{}", e),
+fn main() {
+    match safe_divide(10, 0) {
+            Ok(value) => println!("Division successful: {}", value),
+            Err(e) => println!("{}", e),
     }
+}
 ```
 
-## Error with `panic!` 
-When program encounters an error which is unrecoverable then a `panic!` occures. This will stop execution and helpful for the case where continuing execution doesn't make any sense.
+## Error with `panic` 
+When program encounters an error which is unrecoverable then a `panic` occurs. This will stop the execution, which is helpful in the case where continuing the execution wouldn't make any sense.
 
-> Example
-```
++ Example
+```rust
 fn main() {
+    let mut data = array![1, 2];
+
     if true {
-        panic!("2");
+        panic(data);
     }
-    println!("This line isn't reached");
+    println!("This line will not be reached.");
 }
 ```
-> **Output** 
```

Comment:
Hmm, maybe add a line like:
``````markdown
Running this example would print this in the console:

```console
Run panicked with [1, 2, ].
```